src/openvic-simulation/map/Map.cpp

```diff
@@ -529,21 +536,28 @@ bool Map::load_map_images(fs::path const& province_path, fs::path const& terrain
 					);
 				}
 			}
+
 			province_shape_image[idx].index = Province::NULL_INDEX;
 
-		set_terrain:
-			const TerrainTypeMapping::index_t terrain = terrain_data[idx];
-			TerrainTypeMapping const* mapping = terrain_type_manager.get_terrain_type_mapping_for(terrain);
-			if (mapping != nullptr) {
-				if (province_shape_image[idx].index != Province::NULL_INDEX) {
-					terrain_type_pixels_list[province_shape_image[idx].index - 1][&mapping->get_type()]++;
+			set_terrain:
+				const Province::index_t province_index = get_index_from_colour(province_colour);
```

Comment:
You already have the province index `province_shape_image[idx].index`, no need to look up by colour again.

Reply:
idx was all I needed, so I use that now.